tests/rptest/tests/archival_test.py

```diff
@@ -155,7 +155,7 @@ class ArchivalTest(RedpandaTest):
 
     def __init__(self, test_context):
         self.s3_bucket_name = f"panda-bucket-{uuid.uuid1()}"
-        extra_rp_conf = dict(
+        self._extra_rp_conf = dict(
             developer_mode=True,
```

Comment:
nit: Consider removing this line since developer mode is enabled by default now.